components/console-backend-service/internal/domain/eventing/trigger/converter.go

```diff
@@ -0,0 +1,189 @@
+package trigger
+
+import (
+	"errors"
+	"fmt"
+
+	"knative.dev/pkg/apis"
+
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+//go:generate mockery -name=GQLConverter -output=automock -outpkg=automock -case=underscore
+//go:generate failery -name=GQLConverter -case=underscore -output disabled -outpkg disabled
+type GQLConverter interface {
+	ToGQL(in *v1alpha1.Trigger) (*gqlschema.Trigger, error)
+	ToGQLs(in []*v1alpha1.Trigger) ([]gqlschema.Trigger, error)
+	ToTrigger(in gqlschema.TriggerCreateInput, ownerRef []gqlschema.OwnerReference) (*v1alpha1.Trigger, error)
+	ToTriggers(in []gqlschema.TriggerCreateInput, ownerRef []gqlschema.OwnerReference) ([]*v1alpha1.Trigger, error)
+}
+
+type triggerConverter struct {
+}
+
+func NewTriggerConverter() GQLConverter {
+	return &triggerConverter{}
+}
+
+func (c *triggerConverter) ToGQL(in *v1alpha1.Trigger) (*gqlschema.Trigger, error) {
+	if in == nil {
+		return nil, errors.New("input trigger cannot be nil")
+	}
+
+	attributes := solveAttributes(in.Spec.Filter)
+	status := calculateStatus(in.Status)
+	dest, err := solveDestination(in.Spec.Subscriber)
+	if err != nil {
+		return nil, err
+	}
+
+	return &gqlschema.Trigger{
+		Name:             in.Name,
+		Namespace:        in.Namespace,
+		Broker:           in.Spec.Broker,
+		FilterAttributes: attributes,
+		Subscriber:       dest,
+		Status:           status,
+	}, nil
+}
+
+func (c *triggerConverter) ToGQLs(in []*v1alpha1.Trigger) ([]gqlschema.Trigger, error) {
+	if in == nil {
+		return nil, errors.New("input triggers cannot be nil")
+	}
+
+	triggers := []gqlschema.Trigger{}
+	for _, trigger := range in {
+		item, err := c.ToGQL(trigger)
+		if err != nil {
+			return nil, err
+		}
+		triggers = append(triggers, *item)
+	}
+	return triggers, nil
+}
+
+func (c *triggerConverter) ToTrigger(in gqlschema.TriggerCreateInput, ownerRef []gqlschema.OwnerReference) (*v1alpha1.Trigger, error) {
+	meta := v1alpha1.SchemeGroupVersion.WithKind("Trigger")
+	trigger := &v1alpha1.Trigger{
+		TypeMeta: v1.TypeMeta{
+			Kind:       meta.Kind,
+			APIVersion: meta.GroupVersion().String(),
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:            in.Name,
+			Namespace:       in.Namespace,
+			OwnerReferences: []v1.OwnerReference{},
+		},
+		Spec: v1alpha1.TriggerSpec{
+			Broker: in.Broker,
+			Filter: solveFilters(in.FilterAttributes),
+		},
+		Status: v1alpha1.TriggerStatus{},
+	}
+
+	for _, ref := range ownerRef {
```

Comment:
You can at the beginning of the function do this operation and add owners to the Trigger object during initialization.